src/libsync/account.cpp

```diff
@@ -201,6 +203,30 @@ void Account::setCredentials(AbstractCredentials *cred)
         this, &Account::slotCredentialsFetched);
     connect(_credentials.data(), &AbstractCredentials::asked,
         this, &Account::slotCredentialsAsked);
+
+    trySetupPushNotifications();
+}
+
+void Account::trySetupPushNotifications()
+{
+    if (_capabilities.availablePushNotifications() & PushNotificationType::Files) {
```

Comment:
Shouldn't it be `_capabilities.availablePushNotifications() != PushNotificationType::None` ?

I mean that part of the setup should be done as soon as any push notification type is supported.

Reply:
Yes, you're right, for now, it would make no difference but in the future, it will be an error.

Reply:
Nope, because "None" has the magic 0 value. :-)

Reply:
Ah yes. But `_capabilities.availablePushNotifications() != PushNotificationType::None` gives me a error. Seems not to be overloaded for these types.

Reply:
Okay I got the mistake. I was declaring PushNotificationType as enum class. If I declare it as enum it works fine. WHat does speak against using testFlag()?

Reply:
Ah right... you end up with this kind of issues when you use enum class for an enum used as a flag. If that's something which needs to be combined somehow and with a 0 value in a way you have values which are combinations and as such not part of the enum really so better have a less strict enum.

If you have a "pure enum" that is a fixed set of values not combined together then enum class is the obvious better choice.

As for testFlag... honestly I'm not 100% sure it works so well with 0, would have to check the implementation.

Reply:
Yeah, I will go for the plain enum